lib/elixir/lib/kernel.ex

```diff
@@ -1810,6 +1810,32 @@ defmodule Kernel do
     do: Access.get_and_update(data, h, &get_and_update_in(&1, t, fun))
 
   @doc """
+  Deletes a key in a nested structure.
+
+  Uses the `Access` protocol to traverse the structures
+  according to the given `keys`, unless the `key` is a
+  function. If the key is a function, it will be invoked
+  as specified in `get_and_update_in/3`.
+
+  ## Examples
+
+      iex> users = %{"john" => %{age: 27}, "meg" => %{age: 23}}
+      iex> delete_in(users, ["john", :age])
+      %{"john" => %{}, "meg" => %{age: 23}}
+
+  In case any entries in the middle returns `nil`,
```

Comment:
One difference between `pop_in` and `delete_in` is that `delete_in` can afford to not fail if any of the entries return nil. This actually makes me more inclined towards `delete_in` because it offers something `update_in` cannot.